modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -77,7 +81,16 @@ object ConsensusManager {
           def getObservedOutcome: F[ConsensusOutcome[Key, Artifact]] = for {
             readyPeers <- clusterStorage.getResponsivePeers
               .map(_.filter(_.state === Ready))
-            selectedPeer <- Random[F].elementOf(readyPeers)
+            numberOfPeersToQuery <- Random[F].nextIntBounded(readyPeers.size / 4 + 1)
+            peersToQuery <- Random[F]
+              .shuffleList(readyPeers.toList)
```

Comment:
Do you realize, that 99% of the time (or more) `readyPeers` will be a collection of size 1?

All this logic isn't going to do anything, except for a very rare case when the single node that we joined is going to fail while we're observing it.

Reply:
@cngo-github we need to ensure this process runs after peer discovery has populated peers, but before download has occurred, this is important change to make here so we should ensure the peer list has been populated by discovery process BEFORE attempting this.

Reply:
Ok, so @cngo-github what changes to the flow are you planning to introduce to make the implementation aligned with what @ryle-goehausen described?